### PR TITLE
AppiumDriverLocalService can handle outputStreams

### PR DESCRIPTION
**Pull request by jiangyitao**
## Change list

Please provide briefly described change list which are you going to propose. 
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Please provide more details about changes if it is necessary. If there are new features you can provide code samples which show the way they
work and possible use cases. Also you can create [gists](https://gist.github.com) with pasted java code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://help.github.com/categories/writing-on-github/)

### CODE REVIEW

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java [hunk 1]
```diff
@@ -294,6 +294,37 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
...
+    }
+
+    /**
+     * Remove the outputStream which is receiving server output data, and close
+     *
+     * @param outputStream
+     * @return true if the outputStream has been removed
+     */
+    public boolean removeOutputStreamAndClose(OutputStream outputStream) throws IOException {
+        boolean removed = removeOutputStream(outputStream);
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:314]: I would rather return the removed stream as Optional, so we could close it in the client code if needed. So these two helpers would not be really necessary

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java [hunk 2]
```diff
@@ -294,6 +294,37 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
         }
     }
 
+    /**
+     * Remove the outputStream which is receiving server output data
+     *
+     * @return true if the outputStream has been removed
+     */
+    public boolean removeOutputStream(OutputStream outputStream) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:302]: Please use the same naming convention as above (Output -> OutPut)

File: src/main/java/io/appium/java_client/service/local/ListOutputStream.java [hunk 3]
```diff
@@ -30,6 +30,10 @@ ListOutputStream add(OutputStream stream) {
         return this;
     }
 
+    boolean remove(OutputStream stream) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/ListOutputStream.java:33]: consider returning `Optional<OutputStream>`  instead

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java [hunk 4]
```diff
@@ -295,33 +296,21 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
...
-        return removed;
-    }
-
     /**
      * Close all existing server output streams.
      *
      * @throws IOException
      */
-    public void closeOutputStreams() throws IOException {
+    public void closeOutPutStreams() throws IOException {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:313]: This method is also not needed now as we could always do something  similar to `removeOutPutStream(x).map(OutputStream::close)`

**Comment by mykola-mokhnach** [comment]
Could you please sign the CLA and fix stylecheck errors?

**Comment by jiangyitao** [comment]
> 
done

**Review by valfirst — APPROVED** [review]

**Pull request merged**